Makefile

```diff
@@ -154,15 +168,17 @@ $(BUILD_DIRECTORY)/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-dependencies/bower
 	| $(BUILD_DIRECTORY)/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-dependencies
 	./scripts/build/dependencies-bower.sh -p -x $|
 
-$(BUILD_DIRECTORY)/electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app: \
+$(BUILD_DIRECTORY)/electron-$(TARGET_PLATFORM)-$(APPLICATION_VERSION)-$(TARGET_ARCH)-app: \
```

Comment:
I guess the downside of changing `electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app` to `electron-$(TARGET_PLATFORM)-$(APPLICATION_VERSION)-$(TARGET_ARCH)-app` (with the `$(APPLICATION_VERSION)` in the default `snapshot` builds containing the git hash), is that every time you do a `make` after changing commits, you'll end up with an entirely separate directory, which kinda destroys the artefact-caching that Make is designed to give you :-(  (and unless you run `make clean` regularly, you'll also end up with a plethora of slightly-differently-named directories in `$(BUILD_DIRECTORY)`)

I know you said that you wanted to keep "as much as possible" in the shell scripts and out of the `Makefile`, but in this circumstance I wonder if a better approach might be to revert the directory names back to `electron-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app` and then have some build-step in the Makefile which force-updates the `version` and `package.json` files as necessary? (in fact, to not screw up the timestamps, it'd probably be better to have a shell script which compared the existing contents of `version`, and didn't update it if it doesn't need to).
I'll have a play and see what I come up with...